filament/backend/src/metal/MetalContext.mm

```diff
@@ -153,5 +153,61 @@ bool isInRenderPass(MetalContext* context) {
     return context->currentRenderPassEncoder != nil;
 }
 
+void MetalPushConstantBuffer::setPushConstant(PushConstantVariant value, uint8_t index) {
+    if (mPushConstants.size() <= index) {
+        mPushConstants.resize(index + 1);
+        mDirty = true;
+    }
+    if (UTILS_LIKELY(mPushConstants[index] != value)) {
+        mDirty = true;
+        mPushConstants[index] = value;
+    }
+}
+
+void MetalPushConstantBuffer::setBytesIfDirty(
+        id<MTLCommandEncoder> encoder, ShaderStage stage) {
+    if (!mDirty) {
+        return;
+    }
+    constexpr size_t PUSH_CONSTANT_SIZE_BYTES = 4;
+    constexpr size_t PUSH_CONSTANT_BUFFER_INDEX = 26;
+
+    size_t bufferSize = PUSH_CONSTANT_SIZE_BYTES * mPushConstants.size();
+    char* buffer = static_cast<char*>(malloc(bufferSize));
+    for (size_t i = 0; i < mPushConstants.size(); i++) {
+        const auto& constant = mPushConstants[i];
+        std::visit([buffer, i](auto arg) {
+                    *(decltype(arg)*)(buffer + PUSH_CONSTANT_SIZE_BYTES * i) = arg;
```

Comment:
how do metal programs interpret bools?  This'll set a 1 in the lowest-word of the 4-bytes?

Reply:
Good question, let me investigate. Maybe it's a good idea to explicitly write all the bits, setting the constant to 0x00000000 or 0x00000001 depending on the bool.

Reply:
So turns out that spirv-cross will convert the push constant `bool`s to `uint`s (probably for alignment purposes). So in that case, I definitely need to ensure I'm writing all the bytes.